java/test.sh

```diff
@@ -7,20 +7,30 @@ set -x
 
 ROOT_DIR=$(cd "$(dirname "${BASH_SOURCE:-$0}")"; pwd)
 
-pushd $ROOT_DIR/../java
-echo "Compiling Java code."
-mvn clean install -Dmaven.test.skip
+# run this file before compile the targets
+sh $ROOT_DIR/generate_deps.sh
```

Comment:
Can this also be a `genrule`? That would be preferable, because one `bazel build ...` command can build everything.